internal/provider/resource_cloudflare_waiting_room_event.go

```diff
@@ -42,19 +42,21 @@ func expandWaitingRoomEvent(d *schema.ResourceData) (cloudflare.WaitingRoomEvent
 		return cloudflare.WaitingRoomEvent{}, err
 	}
 
-	prequeueStartTime := time.Time{}
+	var prequeueStartTime *time.Time
 	if t, ok := d.GetOk("prequeue_start_time"); ok {
-		prequeueStartTime, err = time.Parse(time.RFC3339, t.(string))
+		prequeueStartTimeValue, err := time.Parse(time.RFC3339, t.(string))
+		prequeueStartTime = &prequeueStartTimeValue
 		if err != nil {
 			return cloudflare.WaitingRoomEvent{}, err
 		}
 	}
 
 	return cloudflare.WaitingRoomEvent{
+		ID:                    d.Id(),
```

Comment:
this is going to error in the `Create` where it is referenced as `d.SetId()` isn't called until after the API calls are made and this value doesn't exist.

Reply:
Makes sense, I'll just extract from the ResourceData and set it manually.